Orange/widgets/unsupervised/owsom.py

```diff
@@ -184,6 +187,96 @@ def disconnected_spin(spin):
 N_ITERATIONS = 200
 
 
+class SomSharedValueCompute:
+    def __init__(self, domain: Domain, model: SOM,
+                 offsets: Union[np.ndarray, None],
+                 scales: Union[np.ndarray, None]):
+        # offsets and scales are made immutable so that they are hashable
+        def immutable(a):
+            if a is not None:
+                a = a.copy()
+                a.flags.writeable = False
+            return a
+
+        self.domain = domain
+        self.model = model
+        self.offsets = immutable(offsets)
+        self.scales = immutable(scales)
+        self.__hash = None
```

Comment:
When I cache hashes I try to care that the result is not pickled. Especially if hashing transient things like ids.